OrbitGl/App.cpp

```diff
@@ -571,22 +567,23 @@ bool OrbitApp::StartCapture() {
   int32_t pid = Capture::GTargetProcess->GetID();
   std::map<uint64_t, FunctionInfo*> selected_functions =
       Capture::GSelectedFunctionsMap;
-  thread_pool_->Schedule([this, pid, selected_functions] {
-    capture_client_->Capture(pid, selected_functions);
-  });
+
+  result = capture_client_->StartCapture(thread_pool_.get(), pid,
+                                         selected_functions);
 
   return true;
 }
 
 void OrbitApp::StopCapture() {
-  CHECK(Capture::GState == Capture::State::kStarted);
-  Capture::StopCapture();
-
-  capture_client_->StopCapture();
+  if (!capture_client_->StopCapture()) {
+    LOG("StopCapture failed, likely because it is already stopping/stopped");
```

Comment:
Maybe this LOG (or ERR maybe) can be done inside CaptureClient::StopCapture so it logs why the error has occurred

Reply:
done.